File: pilhas-filas/javascript/stack1.js
export default class Stack1 {
    constructor(size = 100) {
        this.size = size;
        this.arr = new Array(size);
        this.top = -1;
    }

    isEmpty() {
        return this.top === -1;
    }

    isFull() {
        return this.top === this.size - 1;
    }

    push(item) {
        if (this.isFull()) {
            throw new Error("Stack is full");
        }
        this.top++;
        this.arr[this.top] = item;
    }

    pop() {
        if (this.isEmpty()) {
            throw new Error("Stack is empty");
        }
        const item = this.arr[this.top];
        this.top--;
        return item;
    }

    peek() {
        if (this.isEmpty()) {
            throw new Error("Stack is empty");
        }
        return this.arr[this.top];
    }

    count() {
        return this.top + 1;
    }

    clear() {
        this.top = -1;
    }
}

File: pilhas-filas/python/stack1.py
class Stack1:
    def __init__(self, size=100):
        self.size = size
        self.arr = [None] * size
        self.top = -1

    def is_empty(self):
        return self.top == -1

    def is_full(self):
        return self.top == self.size - 1

    def push(self, item):
        if self.is_full():
            raise Exception("Stack is full")
        self.top += 1
        self.arr[self.top] = item

    def pop(self):
        if self.is_empty():
            raise Exception("Stack is empty")
        item = self.arr[self.top]
        self.top -= 1
        return item

    def peek(self):
        if self.is_empty():
            raise Exception("Stack is empty")
        return self.arr[self.top]

    def count(self):
        return self.top + 1

    def clear(self):
        self.top = -1
